ui-spacetimechart/src/stories/additional-data.stories.tsx

```diff
@@ -0,0 +1,291 @@
+      const startPixel = getTimePixel(timeStart);
+      const endPixel = getTimePixel(timeEnd);
+
+      const areaTimeSize = clamp(endPixel, 0, timeSize) - clamp(startPixel, 0, timeSize);
+      if (!inRange(spacePixel, 0, spaceSize) || !areaTimeSize) return;
+
+      // Dashed line:
+      ctx.strokeStyle = ERROR_30;
+      ctx.lineWidth = 2;
+      ctx.setLineDash([4, 4]);
+      ctx.beginPath();
+      if (spaceAxis === 'x') {
+        ctx.moveTo(spacePixel, startPixel);
+        ctx.lineTo(spacePixel, endPixel);
+      } else {
+        ctx.moveTo(startPixel, spacePixel);
+        ctx.lineTo(endPixel, spacePixel);
+      }
+      ctx.stroke();
+      ctx.setLineDash([]);
+
+      // Half discs:
+      const radius = 4;
+      ctx.fillStyle = ERROR_60;
+
+      ctx.beginPath();
+      if (spaceAxis === 'x') {
+        ctx.moveTo(spacePixel, startPixel);
+        ctx.arc(spacePixel, startPixel, radius, Math.PI, Math.PI * 2, true);
+      } else {
+        ctx.moveTo(startPixel, spacePixel);
+        ctx.arc(startPixel, spacePixel, radius, Math.PI / 2, (Math.PI * 3) / 2, true);
+      }
+      ctx.closePath();
+      ctx.fill();
+
+      ctx.beginPath();
+      if (spaceAxis === 'x') {
+        ctx.moveTo(spacePixel, endPixel);
+        ctx.arc(spacePixel, endPixel, radius, Math.PI, Math.PI * 2, false);
+      } else {
+        ctx.moveTo(endPixel, spacePixel);
+        ctx.arc(endPixel, spacePixel, radius, Math.PI / 2, (Math.PI * 3) / 2, false);
+      }
+      ctx.closePath();
+      ctx.fill();
+    },
+    [doorPosition, timeEnd, timeStart]
+  );
+  useDraw('paths', drawClosedDoor);
+
+  return null;
+};
+
+/**
+ * This story aims at showcasing how to display additional data.
+ */
+const Wrapper: FC<{
+  swapAxis: boolean;
+  spaceScaleType: 'linear' | 'proportional';
+}> = ({ swapAxis, spaceScaleType }) => {
+  const [state, setState] = useState<{
+    xOffset: number;
+    yOffset: number;
+    xZoomLevel: number;
+    yZoomLevel: number;
+    panning: null | { initialOffset: Point };
+  }>({
+    xOffset: 0,
+    yOffset: 0,
+    xZoomLevel: X_ZOOM_LEVEL,
+    yZoomLevel: Y_ZOOM_LEVEL,
+    panning: null,
+  });
+
+  return (
+    <div className="inset-0">
+      <SpaceTimeChart
+        className={cx('inset-0 absolute p-0 m-0', state.panning && 'cursor-grabbing')}
+        operationalPoints={OPERATIONAL_POINTS}
+        swapAxis={swapAxis}
+        spaceOrigin={0}
+        spaceScales={OPERATIONAL_POINTS.slice(0, -1).map((point, i) => ({
+          from: point.position,
+          to: OPERATIONAL_POINTS[i + 1].position,
+          ...(spaceScaleType === 'linear'
+            ? { size: 50 * state.yZoomLevel }
+            : { coefficient: 150 / state.yZoomLevel }),
+        }))}
+        timeOrigin={+new Date('2024/04/02')}
+        timeScale={60000 / state.xZoomLevel}
+        xOffset={state.xOffset}
+        yOffset={state.yOffset}
+        onPan={({ initialPosition, position, isPanning }) => {
```

Comment:
Maybe it would be nice to extract this function in a helper since it is used in many stories

Reply:
Agreed.

Reply:
It will be done later.